app/core/versioning.py

```
@@ -0,0 +1,296 @@
+            # Generate schema hash
+            schema_hash = hashlib.sha256(
+                json.dumps(schema, sort_keys=True).encode()
+            ).hexdigest()
+
+            # Create version info
+            version_info = VersionInfo(
+                major=major,
+                minor=minor,
+                patch=patch,
+                status=status,
+                created_at=datetime.utcnow(),
```

Ruff (0.8.2)

`datetime.datetime.utcnow()` used

(DTZ003)